source/core/slang-type-text-util.cpp

```diff
@@ -160,6 +191,54 @@ static const ScalarTypeInfo s_scalarTypeInfo[] =
     return UnownedStringSlice::fromLiteral("unknown");
 }
 
+/* static */SlangCompileTarget TypeTextUtil::asCompileTargetFromExtension(const UnownedStringSlice& slice)
+{
+    if (slice.size())
+    {
+        for (const auto& info : s_compileTargetInfos)
+        {
+            if (StringUtil::indexOfInSplit(UnownedStringSlice(info.extensions), ',', slice) >= 0)
+            {
+                return info.target;
+            }
+        }
+    }
+    return SLANG_TARGET_UNKNOWN;
+}
+
+/* static */ SlangCompileTarget TypeTextUtil::asCompileTargetFromName(const UnownedStringSlice& slice)
+{
+    if (slice.size())
+    {
+        for (const auto& info : s_compileTargetInfos)
+        {
+            if (StringUtil::indexOfInSplit(UnownedStringSlice(info.names), ',', slice) >= 0)
+            {
+                return info.target;
+            }
+        }
+    }
+    return SLANG_TARGET_UNKNOWN;
+}
+
+static Index _getTargetInfoIndex(SlangCompileTarget target)
+{
+    for (Index i = 0; i < SLANG_COUNT_OF(s_compileTargetInfos); ++i)
+    {
+        if (s_compileTargetInfos[i].target == target)
+        {
+            return i;
+        }
+    }
+    return -1;
+}
+
+UnownedStringSlice TypeTextUtil::asNameFromCodeGenTarget(SlangCompileTarget target)
```

Comment:
Similarly, I'd call this just `getCompileTargetName` since this operation can't really fail (or at least it shouldn't - every target should have a canonical name that will be used in diagnostic messages, etc.).

Reply:
Not really related to this change, but `SlangCompileTarget` is poorly named, so having more functions using "compile target" to refer to this type is a bit ugly.

Really the `SlangCompileTarget` enumeration covers a target *format* that the output is encoded in (e.g., DXBC vs. DXIL) which is only one aspect of what defines a compilation target in Slang today. The notion of a compilation target is much closer to the `TargetReq` type at this point, which uses `SlangCompileTarget` only to represent the output format.